lib/runner.js

```diff
@@ -378,7 +379,15 @@ Runner.prototype.hook = function(name, fn) {
       }
       if (err) {
         if (err instanceof Pending) {
-          if (name === HOOK_TYPE_BEFORE_EACH || name === HOOK_TYPE_AFTER_EACH) {
+          if (name === HOOK_TYPE_AFTER_EACH || name === HOOK_TYPE_AFTER_ALL) {
+            utils.deprecate(
+              format(
+                "skipping a test within an '%s' hook is DEPRECATED and will be removed from a future version of Mocha. " +
```

Comment:
1. `name` here is one of `afterAll` or `afterEach`, which aren't intended for humans.  Replace `afterAll` with "after all" and `afterEach` with "after each"
1. instead of "will be removed from" say "will throw an exception in"
1. and "Use a return statement or other means to abort hook execution."

Reply:
to 1.: I have chosen the format `after()` and `afterEach()` which is used mostly in Mocha user documentation. I can change it otherwise.

Reply:
please do; this should be changed across the board, because it's unique to the `bdd` interface

Reply:
done